ops/src/messenger.rs

```diff
@@ -70,18 +72,35 @@ impl Messenger {
         repo: &github::Repo,
         branch: &str,
         commits: &[T],
+        thread_urls: Vec<String>,
     ) {
+        // We could look at the thread_url to see if threads have been used in the past, but that
+        // wouldn't respect the users' choice if they change the setting later.
+        let use_threads = self.config.repos().notify_use_threads(repo) && !thread_urls.clone().is_empty();
```

Comment:
```suggestion
        let use_threads = self.config.repos().notify_use_threads(repo) && !thread_urls.is_empty();
```

should not clone just to check if it's empty.